server/server.go

```diff
@@ -350,26 +350,18 @@ func (server *HTTPServer) deleteClusters(writer http.ResponseWriter, request *ht
 	}
 }
 
-func (server *HTTPServer) addVersionToClusters(orgID types.OrgID, clusters []ctypes.HittingClustersData) ([]types.HittingClustersDataWithVersion, error) {
-	clustersWithVersion := []types.HittingClustersDataWithVersion{}
-
-	for _, cluster := range clusters {
-		version, err := server.Storage.ReadReportInfoForCluster(orgID, cluster.Cluster)
+func (server *HTTPServer) addVersionToClusters(orgID types.OrgID, clusters []ctypes.HittingClustersData) error {
+	for index := range clusters {
+		version, err := server.Storage.ReadReportInfoForCluster(orgID, clusters[index].Cluster)
 
 		if err != nil {
-			return nil, fmt.Errorf("unable to gather version for %s: %w", cluster.Cluster, err)
+			return fmt.Errorf("unable to gather version for %s: %w", clusters[index].Cluster, err)
 		}
 
-		clustersWithVersion = append(
-			clustersWithVersion,
-			types.HittingClustersDataWithVersion{
-				HittingClustersData: cluster,
-				Version:             version,
-			},
-		)
+		clusters[index].Meta = ctypes.ClusterMetadata{Version: version}
```

Comment:
If you want to modify the `clusters` you'll need to pass a pointer

Reply:
AFAIK arrays/slices are passed by reference. If we access the element through its original reference, using the index, it will be modified in the original array.

When iterating over the elements, it gets copies instead, so the modifications are not in the original object

Reply:
@tisnik pleasel could you verify if I'm right?

Reply:
yes, slice is basically struct with three members:
- pointer to an array
- actual length
- capacity